tactics/declare.ml

```diff
@@ -346,6 +346,15 @@ let declare_private_constant ?role ?(local = ImportDefaultBehavior) ~name ~kind
   let eff = { Evd.seff_private = eff; Evd.seff_roles; } in
   kn, eff
 
+let inline_private_constants ~sideff ~univs env ce =
+  let body, eff = Future.force ce.proof_entry_body in
+  let (cb, ctx) =
+    if sideff then Safe_typing.inline_private_constants env (body, eff.Evd.seff_private)
+    else body
```

Comment:
It's very strange to have `inline_private_constants` not inline the constants when called with `sideeff:false`

Reply:
I dunno, that's the previous behavior of `build_by_tactic`, the flag is only relevant on scheme declaration, so I need to research more, but indeed it could well be the case that we can remove it.